source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -202,10 +202,15 @@ void ShaderGraph::addDefaultGeomNode(ShaderInput* input, const GeomPropDef& geom
         // input here and ignore the type of the geomprop. They are required to have the same type.
         string geomNodeDefName = "ND_" + geomprop.getGeomProp() + "_" + input->getType().getName();
         NodeDefPtr geomNodeDef = _document->getNodeDef(geomNodeDefName);
-        if (!geomNodeDef)
+        if (!geomNodeDef && _document->hasDataLibrary())
```

Comment:
This code here looks to me like we're inspecting the local document for a nodeDef first, and then falling back to the dataLibrary.

But then isn't Document::getNodeDef() above already checking the dataLibrary?  so perhaps this additional secondary check isn't necessary?